### PR TITLE
Add option to save the validator log into a file

### PR DESCRIPTION
**Pull request by gabibguedes**
**Issue:** #4011

**Proposed changes**:
- Add an output file into validator
- Add the `--log-file` option into the command

**Status (please check what you already did)**:
- [X] made PR ready for code review
- [X] added some tests for the functionality
- [x] updated the documentation
- [X] updated the changelog
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by msamogh** [comment]
Thanks for the PR, we'll give it a review as soon as possible.

File: tests/core/test_validator.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
         domain_path=DEFAULT_DOMAIN_PATH,
         training_data_paths=[DEFAULT_NLU_DATA, DEFAULT_STORIES_FILE],
     )
-    return await Validator.from_importer(importer)
+    return await Validator.from_importer(importer, out="log.out")
```
Comment (tmbo) [inline tests/core/test_validator.py]: there should be a test which checks the file is created and has the right content

File: rasa/cli/data.py [hunk 2]
```diff
@@ -98,5 +98,7 @@ def validate_files(args):
         domain_path=args.domain, training_data_paths=args.data
     )
 
-    validator = loop.run_until_complete(Validator.from_importer(file_importer))
+    validator = loop.run_until_complete(
```
Comment (tmbo) [inline rasa/cli/data.py]: is there any reason not to directly configure the logger in here instead of in init of the `Validator`?

**Comment by gabibguedes** [comment]
Hi @tmbo I'll move the code to the cli's `data.py` and add the test. Thanks for the review.

**Comment by tmbo** [comment]
Thinking more about this and discussing this with the team, I don't think it makes sense to log to a file from within Rasa. The user should rather either provide a proper python logging configuration or shell pipe the output to a file.

I think what we should do, to make this command more usefull is to fail with a non 0 exit code if there are any issues.

**Pull request closed without merging**